### PR TITLE
Ligand browser / Biased ligand update

### PR DESCRIPTION
**Pull request by AlibekMamyrbekov**
New version of biased ligand browsers
New version of ligand browsers

### CODE REVIEW

File: ligand/models.py [hunk 1]
```diff
@@ -311,188 +339,280 @@ def __str__(self):
...
+    t_value = models.CharField(max_length=60, null=True)
+    log_bias_factor = models.CharField(max_length=60, null=True)
+    log_bias_factor_a = models.CharField(max_length=60, null=True)
+    log_bias_factor_b = models.CharField(max_length=60, null=True)
+    log_bias_factor_c = models.CharField(max_length=60, null=True)
+    log_bias_factor_d = models.CharField(max_length=60, null=True)
+    log_bias_factor_e = models.CharField(max_length=60, null=True)
+    t_factor = models.CharField(max_length=60, null=True)
+    assay_description = models.CharField(max_length=900, null=True)
+    reference_ligand_id = models.CharField(max_length=60, null=True)
```
Comment (AJK-dev) [inline ligand/models.py:534]: Could you change this to a ForeignKey?

File: ligand/models.py [hunk 2]
```diff
@@ -311,188 +339,280 @@ def __str__(self):
...
+    experiment_pathway_distinction = models.CharField(
+        max_length=200, null=True)
+    experiment_system = models.CharField(max_length=40, null=True)
+    experiment_outcome_method = models.CharField(max_length=200, null=True)
+
 
+# Pathways Part - end
+
+
+#New ligand models
```
Comment (AJK-dev) [inline ligand/models.py:568]: Could you create a separate branch for the new ligand models?

File: ligand/models.py [hunk 3]
```diff
@@ -311,188 +339,280 @@ def __str__(self):
...
-    ligand_source_id = models.TextField(null = True)
-    ligand_source_type = models.TextField(null = True)
-
+    endogenous_ligand = models.ForeignKey(
+        Ligand, related_name='endogenous_ligand_bias', on_delete=models.CASCADE,  null=True)
+    auxiliary_protein = models.TextField(null=True)
+    ligand_source_id = models.TextField(null=True)
+    ligand_source_type = models.TextField(null=True)
 
 class ExperimentAssay(models.Model):
```
Comment (AJK-dev) [inline ligand/models.py:421]: Can you possibly rename this to BiasedAssayExperiment to not be confused with the AssayExperiment table

**Review by AJK-dev — COMMENTED** [review]
I'm going to test this locally right now and will add any other comments in a bit.

**Review by AJK-dev — COMMENTED** [review]
Some intermediate comments, pending other commits

File: ligand/models.py (hunk 1 above)
Comment (AlibekMamyrbekov) [reply ligand/models.py]: Done

File: ligand/models.py (hunk 2 above)
Comment (AlibekMamyrbekov) [reply ligand/models.py]: Done

File: ligand/templates/bias_browser_subtypes.html [hunk 4]
```diff
@@ -0,0 +1,1399 @@
...
+			</tr>
+			<tr>
+				<td>Receptors</td>
+				<td id="receptor_counter"></td>
+			</tr>
+		</table>
+	</div>
+
+	<table class="display compact text-nowrap" id='structures_scrollable' style="width: 100%">
+		< <thead id="headers">
```
[the remark this answers is not in the document]
Comment (AlibekMamyrbekov) [reply ligand/templates/bias_browser_subtypes.html:1058]: Done

File: ligand/templates/bias_browser_subtypes.html [hunk 5]
```diff
@@ -0,0 +1,1399 @@
...
+<div style='padding-top: 0px; font-size: 10px; white-space: nowrap; overflow-y:hidden;  width:100%;'>
+	<div class="toolbar_data_css" id="toolbar_data">
+		<table class="display compact text-nowrap toolbar_data_css">
+			<thead>
+				<tr>
+					<h4>Showing unique:</h4>
+				</tr>
+			</thead>
+
+			<tr>
```
[the remark this answers is not in the document]
Comment (AlibekMamyrbekov) [reply ligand/templates/bias_browser_subtypes.html:1046]: Done

**Review by AJK-dev — APPROVED** [review]

**Pull request merged**